### PR TITLE
Removed legacy platform sidecar annotation method

### PR DESCRIPTION
**Pull request by solarkennedy**
This should only be shipped once we are not using it.

### CODE REVIEW

File: pod/annotations_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func TestPodSchemaVersion(t *testing.T) {
 	}
 
 	ver, err := PodSchemaVersion(pod)
-	assert.NilError(t, err)
+	assert.Nil(t, err)
```
Comment (aaronlehmann) [inline pod/annotations_test.go:23]: `assert.NoError`

**Review by aaronlehmann — COMMENTED** [review]
Let's hold off until MD supports sidecars natively.

File: pod/annotations_test.go (hunk 1 above)
Comment (solarkennedy) [reply pod/annotations_test.go:23]: the stretcher testify one doesn't have noError. Pretty sure nil is fine here.

Comment (aaronlehmann) [reply pod/annotations_test.go:23]: ??

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1335

Comment (solarkennedy) [reply pod/annotations_test.go:23]: Whoops!

**Comment by solarkennedy** [comment]
Taken care of in #41

**Pull request closed without merging**